tests/BLUEPRINT/cli/test_indir/SMOKE-TEST_template.json

```diff
@@ -5,2338 +5,1862 @@
         "unit": "N/A",
         "description": null,
         "source": "Input",
-        "mapping": {
-            "PROCESS": {
-                "name": "aspect",
-                "recv": true,
-                "send": true
-            }
-        }
+        "mapping": {}
```

Comment:
I get the mappings here have been blanked by the latest changes in the codes that store some default mappings separately from the parameter definitions. However, I think it would still be good to be able to demonstrate how to apply / override those defaults via a script that generates the input json. As long as we have that somewhere then blanking these is fine, I think - would that mean that these cases would be treated as having no mapping, or would they take the defaults?

Reply:
They would take the defaults at the moment. It is empty here because no external code is actually run before the template is written and the default mappings are only added then. 

I'm in two minds about whether it should still but null here or an empty dict. I suppose they are essentially saying the same thing.

Reply:
Still not quite sure about this. If I have some bluemira inputs that set up a run of another code in a particular way (certain mappings and read/write values) and a default is changed in the code then I may silently end up with a different result. My gut says being explicit here will avoid some confusion down the line, but happy to pick up a bit more of a discussion around this.

Reply:
I've raised #620 to look at a unified way to get the default mappings for an external code onto the ParameterFrame that's being used for a design.